app/src/main/java/com/github/se/orator/model/profile/UserProfileRepositoryFirestore.kt

```diff
@@ -340,6 +350,372 @@ class UserProfileRepositoryFirestore(private val db: FirebaseFirestore) : UserPr
         }
   }
 
+  /**
+   * Get profiles the user received requests from based on the UIDs stored in the user's profile.
+   *
+   * @param recReqUIds List of UIDs of the users the user received requests from to be retrieved.
+   * @param onSuccess Callback to be invoked with the list of friends' profiles.
+   * @param onFailure Callback to be invoked on failure with the exception.
+   */
+  override fun getRecReqProfiles(
```

Comment:
This method is essentially the same as `getFriendsProfiles`. To avoid code duplication, I would suggest creating a common method with the shared code.

Reply:
This is what I initially had done, but when mocking to be able to test, I wanted each method to do different things. This is why I did this